e2e-tests/editor.spec.ts

```diff
@@ -142,44 +142,49 @@ test(
     }
   })
 
-// FIXME: ClipboardItem is not defined when running with this test
-// test('copy & paste block ref and replace its content', async ({ page }) => {
-//   await createRandomPage(page)
-
-//   await page.type('textarea >> nth=0', 'Some random text')
-//   if (IsMac) {
-//     await page.keyboard.press('Meta+c')
-//   } else {
-//     await page.keyboard.press('Control+c')
-//   }
-
-//   await page.pause()
-
-//   await page.press('textarea >> nth=0', 'Enter')
-//   if (IsMac) {
-//     await page.keyboard.press('Meta+v')
-//   } else {
-//     await page.keyboard.press('Control+v')
-//   }
-//   await page.keyboard.press('Escape')
-
-//   const blockRef$ = page.locator('.block-ref >> text="Some random text"');
-
-//   // Check if the newly created block-ref has the same referenced content
-//   await expect(blockRef$).toHaveCount(1);
-
-//   // Edit the last block
-//   await blockRef$.press('Enter')
-
-//   // Move cursor into the block ref
-//   for (let i = 0; i < 4; i++) {
-//     await page.press('textarea >> nth=0', 'ArrowLeft')
-//   }
-
-//   // Trigger replace-block-reference-with-content-at-point
-//   if (IsMac) {
-//     await page.keyboard.press('Meta+Shift+r')
-//   } else {
-//     await page.keyboard.press('Control+Shift+v')
-//   }
-// })
+test('copy & paste block ref and replace its content', async ({ page, block }) => {
+    await createRandomPage(page)
+    let IsWebAPIClipboardSupported = await getIsWebAPIClipboardSupported(page)
+    let promise_capture: Promise<string> = null
+
+    await block.mustFill('Some random text')
+    // FIXME: copy instantly will make content disappear
+    await page.waitForTimeout(1000)
+    if (!IsWebAPIClipboardSupported){
```

Comment:
I think the copy/paste scripts could be wrapped into the fixtures to make copy/paste in text cleaner.
e.g., in `e2e-tests/fixtures.ts`, register the following to the text fixture

```ts
  clipboard = async (page, use) => {
    let isWebAPIClipboardSupported = await getIsWebAPIClipboardSupported(page)
    let _currentCopyItem: string = null

    const copyErrorPrefix = '"Copy without `clipboard-write` permission:"'

    let handler = (msg: ConsoleMessage) => {
      let text = msg.text()
      if (text.startsWith(copyErrorPrefix)) {
        fn(text.substring(copyErrorPrefix.length))
      }
    }

    page.on('console', handler)

    const clipboard = {
      copy: async () => {
        if (IsMac) {
          await page.keyboard.press('Meta+c')
        } else {
          await page.keyboard.press('Control+c')
        }
      },
      paste: async () => {
        if (isWebAPIClipboardSupported) {
          if (IsMac) {
            await page.keyboard.press('Meta+v')
          } else {
            await page.keyboard.press('Control+v')
          }
        } else {
          await block.mustFill(_currentCopyItem)
        }
      },
    }
    await use(clipboard)
    page.removeListener('console', handler)
  }
```

and in tests, we can write something like this:

```ts
test('foo, async ({ page, block, clipboard }) => {
  // ...
  await clipboard.copy();
  // ...
  await clipboard.paste();
}
```

Reply:
My consideration is that `meta+c` is just one of the way in Logseq to copy elements. There are many other interactions with the clipboard-write ability to test.

Reply:
Do you think mocking `navigator.clipboard` directly is a valid approach? I have seen some discussion in https://github.com/cypress-io/cypress/issues/2752

Reply:
@pengx17 I find a point: There's a degraded solution to use `writeText` instead of `write`, in case of missing `ClipboardItem`. So we don't have to emulate that anymore (for now 🤣 )
It just passed the CI.